### ISSUE TITLE
Imprrove table-has-headers check to handle more complex table markups

### ISSUE THREAD

**Issue by pattonwebz** [equalizedigital/accessibility-checker#1153]
Update the table-has-header check to deal with the TODO comment. The TODO should be fully tackled and removed with the changes.

[https://github.com/equalizedigital/accessibility-checker/blob/e0734087cf33f78da38262c7defc9bf926a34142/src/pageScanner/checks/table-has-headers.js#L8](<https://github.com/equalizedigital/accessibility-checker/blob/e0734087cf33f78da38262c7defc9bf926a34142/src/pageScanner/checks/table-has-headers.js#L8>)

A previous attempt was made in a different PR but reverted so it could be tackled in a dedicated PR so it can be better tested and decided on. You may want to take the same approach or a different one.

Commits to reference:<br>9c49dfa539a39cda0ac37d821e96607a2756b60b - Enhanced table-has-headers.js logic with ARIA support, colspan/rowspan handling<br>c4ab7232283e1860a755d296b0f3120c73487988 - Comprehensive test cases (16 additional tests)
